components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceTest.java

```diff
@@ -1,111 +1,233 @@
+        String[] keySecret = oAuthAdminService.registerOAuthConsumer();
+
+        Assert.assertNotNull(keySecret);
+        Assert.assertEquals(keySecret.length, 2);
+        Assert.assertEquals(keySecret[0], CONSUMER_KEY);
+        Assert.assertEquals(keySecret[1], CONSUMER_SECRET);
+    }
+
+    @DataProvider(name = "getDataForAllOAuthApplicationData")
+    public Object[][] getDataForAllOAuthApplicationData() {
+        return new Object[][] { { "admin" }, { null } };
+    }
+
+    @Test(dataProvider = "getDataForAllOAuthApplicationData")
+    public void testGetAllOAuthApplicationData(String userName) throws Exception {
+
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain("carbon.super");
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(-1234);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+
+        whenNew(OAuthAppDAO.class).withNoArguments().thenReturn(oAtuhAppDAO);
+        OAuthAppDO oAuthAppDO = new OAuthAppDO();
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        oAuthAppDO.setApplicationName("testapp1");
+        oAuthAppDO.setUser(authenticatedUser);
+        authenticatedUser.setUserName(userName);
+        when(oAtuhAppDAO.getOAuthConsumerAppsOfUser(userName, -1234)).thenReturn(new OAuthAppDO[] { oAuthAppDO });
+        OAuthAdminService oAuthAdminService = new OAuthAdminService();
+        try {
+            OAuthConsumerAppDTO[] allOAuthApplicationData =
+                    oAuthAdminService.getAllOAuthApplicationData();
+            Assert.assertNotNull(allOAuthApplicationData);
+            Assert.assertEquals(allOAuthApplicationData.length, 1);
+            Assert.assertEquals(allOAuthApplicationData[0].getApplicationName(), "testapp1");
+        } catch (IdentityOAuthAdminException allOAuthApplicationData) {
+            Assert.assertEquals(allOAuthApplicationData.getMessage(),
+                                "User not logged in to get all registered OAuth Applications");
+        }
+    }
+
+
+    @DataProvider(name = "getRegisterOAuthApplicationData")
+    public Object[][] getRegisterOAuthApplicationData() {
+        return new Object[][] { { OAuthConstants.OAuthVersions.VERSION_2, "admin" },
+                                { OAuthConstants.OAuthVersions.VERSION_2, null },
+                                { null, "admin" }
+        };
+    }
+
+    @Test(dataProvider = "getRegisterOAuthApplicationData")
+    public void testRegisterOAuthApplicationData(String oauthVersion, String userName) throws Exception {
+
+
+        initConfigsAndRealm();
+
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain("carbon.super");
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(-1234);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUserRealm(userRealm);
```

Comment:
this should be in a separate util method. We already have one I suppose. Then, wrap it with try-finally block. The finally block should end the tenant flow.